src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -232,13 +232,13 @@ public function searchBarAction(Request $request)
             $limit =
             $token = null;
         } else {
-            $ip = $session->get('_profiler_search_ip');
-            $method = $session->get('_profiler_search_method');
-            $url = $session->get('_profiler_search_url');
-            $start = $session->get('_profiler_search_start');
-            $end = $session->get('_profiler_search_end');
-            $limit = $session->get('_profiler_search_limit');
-            $token = $session->get('_profiler_search_token');
+            $ip = $request->get('ip', $session->get('_profiler_search_ip'));
```

Comment:
exactly (even though the implementation has been optimized a but to avoid relying on chaining defaults, making it OK when you actually need to allow the 3 sources)